src/vkscript/parser.cc

```diff
@@ -176,6 +183,52 @@ Result Parser::Parse(const std::string& input) {
   return {};
 }
 
+Result Parser::GenerateDefaultPipeline(SectionParser* section_parser) {
```

Comment:
`const SectionParser& section_parser` would be better?

Reply:
I did that originally, but then I run into const'ness issues later on. This makes it work for now we can dig into const'ness later.

Reply:
I do not have strong opinion on this so it is fine but the following patch works fine on my laptop.
Please adopt it if you think it is fine:

```
diff --git a/src/vkscript/parser.cc b/src/vkscript/parser.cc
index 1e0415c..1e88e4e 100644
--- a/src/vkscript/parser.cc
+++ b/src/vkscript/parser.cc
@@ -170,7 +170,7 @@ Result Parser::Parse(const std::string& input) {
   if (!r.IsSuccess())
     return r;
 
-  r = GenerateDefaultPipeline(&section_parser);
+  r = GenerateDefaultPipeline(section_parser);
   if (!r.IsSuccess())
     return r;
 
@@ -183,10 +183,10 @@ Result Parser::Parse(const std::string& input) {
   return {};
 }
 
-Result Parser::GenerateDefaultPipeline(SectionParser* section_parser) {
+Result Parser::GenerateDefaultPipeline(const SectionParser& section_parser) {
   // Generate a pipeline for VkScript.
   PipelineType pipeline_type = PipelineType::kCompute;
-  for (const auto& section : section_parser->Sections()) {
+  for (const auto& section : section_parser.Sections()) {
     if (!SectionParser::HasShader(section.section_type))
       continue;
 
diff --git a/src/vkscript/parser.h b/src/vkscript/parser.h
index a48e605..70e30d1 100644
--- a/src/vkscript/parser.h
+++ b/src/vkscript/parser.h
@@ -38,7 +38,7 @@ class Parser : public amber::Parser {
 
  private:
   std::string make_error(const Tokenizer& tokenizer, const std::string& err);
-  Result GenerateDefaultPipeline(SectionParser* section_parser);
+  Result GenerateDefaultPipeline(const SectionParser& section_parser);
   Result ProcessSection(const SectionParser::Section& section);
   Result ProcessShaderBlock(const SectionParser::Section& section);
   Result ProcessRequireBlock(const SectionParser::Section& section);
diff --git a/src/vkscript/section_parser.h b/src/vkscript/section_parser.h
index 3acecab..3d898c3 100644
--- a/src/vkscript/section_parser.h
+++ b/src/vkscript/section_parser.h
@@ -50,7 +50,7 @@ class SectionParser {
   ~SectionParser();
 
   Result Parse(const std::string& data);
-  const std::vector<Section>& Sections() { return sections_; }
+  const std::vector<Section>& Sections() const { return sections_; }
 
   Result SplitSectionsForTesting(const std::string& data) {
     return SplitSections(data);
```

Reply:
Hah, for some reason I thought Sections was const already, so when I got the build error thought something else was going wrong. Thanks for digging into it, will update this CL.

Reply:
done, thanks.